xarray/core/types.py

```diff
@@ -281,4 +281,5 @@ def copy(
 ]
 
 
+NetcdfWriteModes = Literal["w", "a"]
```

Comment:
Hopefully this looks like an okay change. I liked the equivalent `ZarrWriteModes` and thought I'd adopt it for NetCDF, too. (Can revert if preferred)

Reply:
Ah ha - I've opened myself up to comments on all this! The change here only consolidates the typing for the modes that are applied throughout the code, without changing the selection of values. (As in now `Literal["w", "a"]` is written in one place, rather than many - so if changing the possible values is desired, it would be easier to do at least)

Reply:
So, these are *write* modes, so I think you are correct.  Only 'x' could be added, but /shrug?  carry on.